File: ui/src/main/java/com/dastanapps/dastanlib/customUI/CircleView.java
package com.dastanapps.dastanlib.customUI;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.util.AttributeSet;
import android.view.View;

/**
 * An ImageView that draws its contents as a circle.
 * <p>
 * Normally you should probably make a custom Drawable, but this is just for explanatory purposes.
 */
public class CircleView extends View {

    private Paint mPaint;
    private Paint mStrokePaint;

    public CircleView(Context context, AttributeSet attrs) {
        super(context, attrs);
        mPaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        mStrokePaint = new Paint(Paint.ANTI_ALIAS_FLAG);
        mStrokePaint.setAntiAlias(true);
        mStrokePaint.setStyle(Paint.Style.STROKE);
        mStrokePaint.setStrokeWidth(2);
        mStrokePaint.setColor(Color.BLACK);
        mStrokePaint.setStrokeCap(Paint.Cap.BUTT);

    }

    public void setStrokeWidth(float width) {
        mStrokePaint.setStrokeWidth(width);
    }

    public void setStrokeColor(String color) {
        mStrokePaint.setColor(Color.parseColor(color));
    }

    public void setColor(String color) {
        mPaint.setColor(Color.parseColor(color));
    }


    @Override
    protected void onDraw(Canvas canvas) {
        int width = getWidth();
        int height = getHeight();

        canvas.drawCircle(width / 2, height / 2, (width - 10) / 2, mPaint);
        canvas.drawCircle(width / 2, height / 2, (width - 10) / 2, mStrokePaint);
    }
}

File: ui/src/main/java/com/dastanapps/dastanlib/customUI/CircleView.kt
package com.dastanapps.dastanlib.customUI

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.util.AttributeSet
import android.view.View

/**
 * An ImageView that draws its contents as a circle.
 *
 *
 * Normally you should probably make a custom Drawable, but this is just for explanatory purposes.
 */
class CircleView(context: Context, attrs: AttributeSet) : View(context, attrs) {

    private val mPaint: Paint
    private val mStrokePaint: Paint

    init {
        mPaint = Paint(Paint.ANTI_ALIAS_FLAG)
        mStrokePaint = Paint(Paint.ANTI_ALIAS_FLAG)
        mStrokePaint.isAntiAlias = true
        mStrokePaint.style = Paint.Style.STROKE
        mStrokePaint.strokeWidth = 2f
        mStrokePaint.color = Color.BLACK
        mStrokePaint.strokeCap = Paint.Cap.BUTT

    }

    fun setStrokeWidth(width: Float) {
        mStrokePaint.strokeWidth = width
    }

    fun setStrokeColor(color: String) {
        mStrokePaint.color = Color.parseColor(color)
    }

    fun setColor(color: String) {
        mPaint.color = Color.parseColor(color)
    }


    override fun onDraw(canvas: Canvas) {
        val width = width
        val height = height

        canvas.drawCircle((width / 2).toFloat(), (height / 2).toFloat(), ((width - 10) / 2).toFloat(), mPaint)
        canvas.drawCircle((width / 2).toFloat(), (height / 2).toFloat(), ((width - 10) / 2).toFloat(), mStrokePaint)
    }
}
